components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AMDefaultKeyManagerImpl.java

```diff
@@ -565,7 +573,13 @@ public OAuthApplicationInfo mapOAuthApplication(OAuthAppRequest appInfoRequest)
         //check whether given consumer key and secret match or not. If it does not match throw an exception.
         ClientInfo clientInfo;
         try {
-            clientInfo = dcrClient.getApplication(consumerKey);
+            encoded = false;
+            String isConsumerKeyEncoded = System.getProperty(ENCODE_CONSUMER_KEY, "false");
+            if (isConsumerKeyEncoded.equalsIgnoreCase("true")) {
```

Comment:
Same here

Reply:
The default value is set to be "false" in the above method to get the system property, hence it can not become null

Reply:
Addressed in b1a9d7e66f2d8ee05ee815b7e2d632dc91eeb734